crypto/evp/digest.c

```diff
@@ -885,7 +885,7 @@ static int evp_md_cache_constants(EVP_MD *md)
     ok = evp_do_md_getparams(md, params);
     if (mdsize > INT_MAX || blksz > INT_MAX)
         ok = 0;
-    if (ok) {
+    if (ok > 0) {
```

Comment:
The caller is also  .. if (!evp_md_cache_constants(md))

Reply:
Fixed.